suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/RemoveRolloutSuppressions.java

```diff
@@ -0,0 +1,164 @@
+                // In this case, we actually want to remove all the suppressions
+                .filter(s -> !s.isEmpty())
+                .map(s -> CommonConstants.AUTOMATICALLY_ADDED_PREFIX + s)
+                .collect(Collectors.toSet());
+
+        List<String> existingSuppressions =
+                AnnotationUtils.annotationStringValues(tree).collect(Collectors.toList());
+
+        final List<String> updatedSuppressions;
+        if (suppressionsToRemove.isEmpty()) {
+            // We want to remove all automated suppressions if no specific argument is passed
+            updatedSuppressions = existingSuppressions.stream()
+                    .filter(suppression -> !suppression.startsWith(CommonConstants.AUTOMATICALLY_ADDED_PREFIX))
+                    .collect(Collectors.toList());
+        } else {
+            updatedSuppressions = existingSuppressions.stream()
+                    .filter(suppression -> !suppressionsToRemove.contains(suppression))
+                    .collect(Collectors.toList());
+        }
+
+        if (existingSuppressions.size() == updatedSuppressions.size()) {
+            return Description.NO_MATCH;
+        }
+
+        String updatedText = SuppressWarningsUtils.suppressWarningsString(updatedSuppressions);
+
+        return buildDescription(tree)
+                .addFix(new LineRemovingReplacementFix(state.getSourceCode(), (DiagnosticPosition) tree, updatedText))
+                .build();
+    }
+
+    /**
+     * This class has been introduced because the normal {@link com.google.errorprone.fixes.SuggestedFix} does not
+     *   allow us to introduce a Replacement which has a different start position than the tree's normal start position.
+     * Here we want to:
+     *   - replace the element defined by the provided position
+     *   - also replace the whitespace before the element, up to and including the newline
+     * This way, if e.g. @SuppressWarnings("foo") must be removed entirely, we can remove the entire line, rather than
+     *   just the annotation, leaving us with an empty line.
+     *
+     * Note that this will only delete the whitespace before the element if the entire element is removed
+     *   (i.e. if the replacement text is null or empty).
+     */
+    private static final class LineRemovingReplacementFix implements Fix {
+        private final CharSequence sourceCode;
+        private final DiagnosticPosition position;
+        private final String replacementText;
+
+        private LineRemovingReplacementFix(
+                CharSequence sourceCode, DiagnosticPosition position, String replacementText) {
+            this.sourceCode = sourceCode;
+            this.position = position;
+            // Guarantee replacementText isn't empty to simplify the checks below
+            this.replacementText = replacementText == null ? "" : replacementText;
```

Comment:
Would `replacementText` ever be null?

Reply:
Not in our case actually, so I can simplify the code here indeed.